source/solvers/gls_sharp_navier_stokes.cc

```diff
@@ -81,9 +81,35 @@ template <int dim>
 void
 GLSSharpNavierStokesSolver<dim>::define_particles()
 {
-  particles = this->simulation_parameters.particlesParameters.particles;
-  table_f.resize(particles.size());
-  table_t.resize(particles.size());
+    particles = this->simulation_parameters.particlesParameters.particles;
+    table_f.resize(particles.size());
+    table_t.resize(particles.size());
+
+    for (unsigned int p = 0; p < particles.size(); ++p) {
+
+        particles[p].forces[0]=0;
+        particles[p].forces[1]=0;
+        particles[p].last_forces[0]=0;
+        particles[p].last_forces[1]=0;
+        particles[p].last_torques=0;
+        if (dim==3) {
+            particles[p].forces[2] = 0;
+            particles[p].last_forces[2] = 0;
+        }
+        particles[p].last_velocity=particles[p].velocity;
+        particles[p].velocity_iter=particles[p].velocity;
+        particles[p].last_position=particles[p].position;
+
+        particles[p].last_omega=particles[p].omega;
+        particles[p].omega_iter=particles[p].omega;
+        particles[p].angular_position[0]=0;
+        particles[p].angular_position[1]=0;
+        particles[p].angular_position[2]=0;
+        particles[p].last_angular_position=particles[p].angular_position;
+        particles[p].local_alpha_torque=1;
+        particles[p].local_alpha_force=1;
```

Comment:
Why not create a default constructor for your particle class instead that sets those value? It would be a lot better to have a default constructor than to rely on the solver itself to set the particles. Encapsulation young man!